modules/tensor_mechanics/src/actions/TensorMechanicsRSphericalAction.C

```diff
@@ -0,0 +1,75 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "TensorMechanicsRSphericalAction.h"
+
+#include "Factory.h"
+#include "FEProblem.h"
+#include "Parser.h"
+
+template<>
+InputParameters validParams<TensorMechanicsRSphericalAction>()
+{
+  InputParameters params = validParams<Action>();
+  params.addClassDescription("Set up stress divergence kernel for the 1D spherical problem");
+  params.addRequiredParam<std::vector<NonlinearVariableName> >("displacements", "The nonlinear displacement variable for the problem (should only be a single variable)");
+  params.addParam<std::string>("base_name", "Material property base name");
+  params.addParam<bool>("use_displaced_mesh", false, "Whether to use displaced mesh in the kernels");
+
+  params.addParam<std::vector<AuxVariableName> >("save_in_disp_r", "Auxiliary variables to save the r displacement residuals.");
+  return params;
+}
+
+TensorMechanicsRSphericalAction::TensorMechanicsRSphericalAction(const InputParameters & params) :
+    Action(params)
+{
+}
+
+void
+TensorMechanicsRSphericalAction::act()
+{
+  std::vector<NonlinearVariableName> displacements = getParam<std::vector<NonlinearVariableName> > ("displacements");
+  std::vector<VariableName> coupled_displacements;
+  Real dim = displacements.size();
```

Comment:
for dim you mean?  If yes, I'll fix this here and in the AxisymmetricRZ action as well.